api/src/server.rs

```diff
@@ -340,641 +339,15 @@ where
     }
 }
 
-fn decode_event_id(value: &str) -> Result<EventId, ApiError> {
+pub(crate) fn decode_event_id(value: &str) -> Result<EventId, ApiError> {
```

Comment:
why did these need to become crate public?

Reply:
they are used by the tests which can't access them from the new file. I haven't really looked at whether that means there could be something semi-public that wraps this better (i.e. an input validator). same answer for `decode_event_data`